libs/chain-signatures/contract/tests/back_compatibility.rs

```diff
@@ -117,3 +118,68 @@ async fn test_back_compatiblity_mainnet() {
 async fn test_back_compatiblity_testnet() {
     back_compatibility(Network::Testnet).await.unwrap();
 }
+
+#[tokio::test]
+async fn participant_set_is_unchanged_during_upgrade() -> anyhow::Result<()> {
+    let worker = near_workspaces::sandbox().await?;
+    let contract = deploy_old(&worker, Network::Testnet).await?;
+
+    init_contract(worker, &contract).await?;
+
+    let initial_participants = get_participants(&contract).await?;
+
+    let contract = upgrade_to_new(contract).await?;
+
+    migrate(&contract)
+        .await
+        .expect("❌ Back compatibility check failed: migration() failed");
+
+    let participants_after_upgrade = get_participants(&contract).await?;
+    assert_eq!(
+        initial_participants, participants_after_upgrade,
+        "Participant set must not change after an upgrade."
+    );
+
+    Ok(())
+}
+
+#[tokio::test]
+async fn test_all_participants_have_valid_attestation_for_soft_launch() -> anyhow::Result<()> {
+    let worker = near_workspaces::sandbox().await?;
+    let contract = deploy_old(&worker, Network::Testnet).await?;
+
+    init_contract(worker, &contract).await?;
+
+    let initial_participants = get_participants(&contract).await?;
+    let participant_set_is_not_empty = !initial_participants.participants().is_empty();
+    assert!(
+        participant_set_is_not_empty,
+        "Test must contain a contract with at least one participant"
+    );
```

Comment:
Any particular reason you added this `assert` only to this test and not to the `participant_set_is_unchanged_during_upgrade` test as well?